airflow/dag_processing/manager.py

```diff
@@ -820,10 +821,12 @@ def clear_nonexistent_import_errors(file_paths: list[str] | None, session=NEW_SE
         :param file_paths: list of paths to DAG definition files
         :param session: session for ORM operations
         """
-        query = session.query(errors.ImportError)
         if file_paths:
-            query = query.filter(~errors.ImportError.filename.in_(file_paths))
-        query.delete(synchronize_session="fetch")
+            query = delete(errors.ImportError).where(~errors.ImportError.filename.in_(file_paths))
+        else:
+            query = delete(errors.ImportError)
```

Comment:
you can define the delete before the if, and add the where in the if block

Reply:
done